### PR TITLE
planner: add error message for group_concat

### LINKED ISSUE #10996: window function: group_concat error message is not compatible
## Description

Please answer these questions before submitting your issue. Thanks!

1. What did you do?
If possible, provide a recipe for reproducing the error.

```mysql
mysql> select group_concat('3') over ();
ERROR 1064 (42000): You have an error in your SQL syntax; check the manual that corresponds to your TiDB version for the right syntax to use line 1 column 29 near "over ()"
```

2. What did you expect to see?

```
mysql> select group_concat('3') over ();
ERROR 1235 (42000): This version of MySQL doesn't yet support 'group_concat as window function'
```

## Score

* 50

## Mentor(s)

* @SunRunAway 

Contact the mentors: **#tidb-challenge-program** channel in [TiDB Community](https://join.slack.com/t/tidbcommunity/shared_invite/enQtNzc0MzI4ODExMDc4LWYwYmIzMjZkYzJiNDUxMmZlN2FiMGJkZjAyMzQ5NGU0NGY0NzI3NTYwMjAyNGQ1N2I2ZjAxNzc1OGUwYWM0NzE)  Slack Workspace

## Recommended Skills

* Golang Programming.
* Be familiar with window function and TiDB Parser.

## Learning Materials

* Chinese: [TiDB 精选技术讲解文章](https://github.com/pingcap/presentations/blob/master/hackathon-2019/reference-document-of-hackathon-2019.md)
* English: [Awesome-Database-Learning](https://github.com/pingcap/awesome-database-learning)

### PR DESCRIPTION
**Pull request by zhaox1n**
UCP #10996 

### What problem does this PR solve? 

close #10996, compatible with mysql error messages when using group_cancat as window function. details ref #10996

### Check List 

Tests 

 - Unit test

### CODE REVIEW

**Comment by zhaox1n** [comment]
PTAL @SunRunAway

File: planner/core/testdata/plan_suite_unexported_in.json [hunk 1]
```diff
@@ -264,7 +264,9 @@
       "SELECT SUM(a) OVER w AS 'sum' FROM t WINDOW w AS (ROWS BETWEEN CURRENT ROW AND 1 PRECEDING )",
       "SELECT SUM(a) OVER w AS 'sum' FROM t WINDOW w AS (ROWS BETWEEN 1 FOLLOWING AND 1 PRECEDING )",
       // Test issue 11943
-      "SELECT ROW_NUMBER() OVER (partition by b) + a FROM t"
+      "SELECT ROW_NUMBER() OVER (partition by b) + a FROM t",
+	  // Test issue 10996
+	  "SELECT GROUP_CONCAT(a) OVER () FROM t"
```
Comment (SunRunAway) [inline planner/core/testdata/plan_suite_unexported_in.json]: Please use space as indent and make them align.

File: planner/core/testdata/plan_suite_unexported_out.json [hunk 2]
```diff
@@ -243,7 +243,8 @@
       "[planner:3586]Window 'w': frame start or end is negative, NULL or of non-integral type",
       "[planner:3586]Window 'w': frame start or end is negative, NULL or of non-integral type",
       "[planner:3586]Window 'w': frame start or end is negative, NULL or of non-integral type",
-      "TableReader(Table(t))->Sort->Window(row_number()->Column#14 over(partition by test.t.b))->Projection"
+      "TableReader(Table(t))->Sort->Window(row_number()->Column#14 over(partition by test.t.b))->Projection",
+	  "[planner:1235]This version of TiDB doesn't yet support 'group_concat as window function'"
```
Comment (SunRunAway) [inline planner/core/testdata/plan_suite_unexported_out.json]: ditto.

Comment (zhaox1n) [reply planner/core/testdata/plan_suite_unexported_out.json]: @SunRunAway done

Comment (zhaox1n) [reply planner/core/testdata/plan_suite_unexported_out.json]: @SunRunAway PTAL

**Pull request merged**